dbt/include/spark/macros/materializations/table.sql

```diff
@@ -14,16 +14,22 @@
   -- setup: if the target relation already exists, drop it
   -- in case if the existing and future table is delta or iceberg, we want to do a
   -- create or replace table instead of dropping, so we don't have the table unavailable
-  {% if old_relation and not (old_relation.is_delta and config.get('file_format', validator=validation.any[basestring]) == 'delta') -%}
-    {{ adapter.drop_relation(old_relation) }}
-  {%- endif %}
-
-  {% if old_relation and not (old_relation.is_iceberg and config.get('file_format', validator=validation.any[basestring]) == 'iceberg') -%}
-    {{ adapter.drop_relation(old_relation) }}
-  {%- endif %}
+  {% if old_relation is not none %}
+    {% set is_delta = (old_relation.is_delta and config.get('file_format', validator=validation.any[basestring]) == 'delta') %}
+    {% set is_iceberg = (old_relation.is_iceberg and config.get('file_format', validator=validation.any[basestring]) == 'iceberg') %}
+    {% set old_relation_type = old_relation.type %}
+  {% else %}
+    {% set is_delta = false %}
+    {% set is_iceberg = false %}
+    {% set old_relation_type = target_relation.type %}
+  {% endif %}
+
+  {% if not is_delta and not is_iceberg %}
+    {% set existing_relation = target_relation %}
+    {{ adapter.drop_relation(existing_relation.incorporate(type=old_relation_type)) }}
```

Comment:
I don't know what this means but would like to

Reply:
We ran into an issue where we couldn't always see the `old_relation` on a non-admin account. So I dropped the `target_relation` instead of the `old_relation`. If `old_relation` exists, they are always named the same. The thing that could vary is the type, e.g. a view is getting replaced by a table. So I pull the `old_relation` type onto the `target_relation`. However, I don't want to change the `target_relation`. So I pull a copy of it and adjust the type on that to drop it.